### PR TITLE
Turbo-ize

### PR DESCRIPTION
**Pull request by lishaduck**
Depends on #163
Relates to #156

npm-run-all kept acting up, so I went ahead and migrated elm-review to turbo.
Key things: everything is parallelized now. It's also cached.

To run tasks, use `turbo ...` instead of `npm run ...`, which is installable via `npm install -g turbo`
You can also use `npx turbo`. They both run using _our_ pinned turbo, the global one just proxies to the local one.
And, of course, you can still run `npm run ...`, it just won't get cached.

If you want to add a task, just add it in the package.json. It'll automatically get cached. If you find it gets invalidated too much, you can add it to `turbo.json`.

Here's the diff as GH doesn't play well with stacks from forks: <https://github.com/lishaduck/node-elm-review/compare/bump-deps-2...turbo>
I also updated CI. I hope you don't mind.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -9,16 +9,13 @@
   "packageManager": "npm@10.7.0",
   "scripts": {
     "elm-format": "elm-format --validate ast-codec init-templates new-package parseElm review template",
-    "test": "npm-run-all --print-name --silent elm-tests elm-format tsc eslint-check prettier-check test-run",
```
Comment (jfmengels) [inline package.json:10]: Do you know if it's possible to keep a test script so that I can just run `npm test` ?

**Review by jfmengels — APPROVED** [review]
Thank you for the PR. I'm willing to try `turbo` out.

Can you tell me where I can find the docs for turbo? I'm having trouble finding it.

Do you also know if there's a way to not have the interactive window showing up? Mostly out of curiosity.

> npm-run-all kept acting up

Do you remember what kind of problems you were encountering?

**Comment by lishaduck** [comment]
> ~Can you tell me where I can find the docs for turbo? I'm having trouble finding it.~ It's at https://turbo.build/repo/docs I kept getting confused because I though `turbo` was a different tool than `turborepo`.

Yeah. turbo is the cli, which came first. Then they made turbopack, so they retroactively named it turborepo, but they share a cli (well, they will eventually).

> Do you also know if there's a way to not have the interactive window showing up? Mostly out of curiosity.

The `--log-order` flag. You can pass either --log-order=grouped or --log-order=stream.
stream is essentially npm-run-all

> > npm-run-all kept acting up
> 
> Do you remember what kind of problems you were encountering?

It wasn't injecting node_modules/.bin into the path, so some tests were failing.

> Do you know if it's possible to keep a test script so that I can just run `npm test` ?

Yeah. I can rename `test` to, say, `testing`, and then alias `npm test` to `turbo testing`. It's meant to be used in a monorepo where you do that for all your tasks in your root, then use turbo in the individual packages, but it works well in single package repos as well.

**Comment by jfmengels** [comment]
> I can rename test to, say, testing, and then alias npm test to turbo testing

I think this would be great. `npm test` (or actually `npm t`) is a very ingrained shortcut for me :smile:

**Comment by lishaduck** [comment]
If you prefer the log-order flags, I just found out you can set an env variable:
Set `export TURBO_UI=0` and `export TURBO_LOG_ORDER=streamed` (or `grouped`) in your `.zshrc` (or `.bash_profile` or whatever)

**Pull request merged**